docs/thehive/setup/operations/backup-restore.md

```diff
@@ -104,85 +113,106 @@ Considering that your keyspace is `thehive` and `backup_name` is the name of the
-    ## and an tbz archive containing the snapshot
+    ## Create an tbz archive containing the snapshot
 
     ## Complete variables before running:
     ## KEYSPACE: Identify the right keyspace to save in cassandra
     ## SNAPSHOT: choose a name for the backup
 
-    IP=10.1.1.1
-    SOURCE_KEYSPACE=thehive
-    SNAPSHOT=thehive_20211124
-    SNAPSHOT_INDEX=1
+    SOURCE_KEYSPACE="thehive"
+    SNAPSHOT="thehive-backup"
+    SNAPSHOT_DATE="$(date +%F)"
 
     # Backup Cassandra
-
     nodetool cleanup ${SOURCE_KEYSPACE}
 
-    nodetool snapshot ${SOURCE_KEYSPACE}  -t ${SNAPSHOT}_${SNAPSHOT_INDEX}
-
-    echo -n "Cassandra admin password":
-    read -s CASSANDRA_PASSWORD
+    nodetool snapshot ${SOURCE_KEYSPACE}  -t ${SNAPSHOT}_${SNAPSHOT_DATE}
 
-    ## Save schema
-    cqlsh -u cassandra -p ${CASSANDRA_PASSWORD} ${IP} -e "DESCRIBE KEYSPACE ${SOURCE_KEYSPACE}" > schema_${SNAPSHOT}_${SNAPSHOT_INDEX}.cql
-
-    ## Create archive
-    if [[ $? == 0 ]]
-    then
-    tar cjf ${SNAPSHOT}_${SNAPSHOT_INDEX}.tbz /var/lib/cassandra/data/${SOURCE_KEYSPACE}/*/snapshots/${SNAPSHOT}_${SNAPSHOT_INDEX}/
-    fi
+    tar cjf ${SNAPSHOT}_${SNAPSHOT_DATE}.tbz /var/lib/cassandra/data/${SOURCE_KEYSPACE}/*/snapshots/${SNAPSHOT}_${SNAPSHOT_DATE}/
     ```
 
 ### Restore data
 
+#### Data and index
+
+Before starting TheHive, if there is an index, ensure the index in Elasticsearch match the data in Cassandra otherwise this would imply unpredictive behavior. You can force resync (reindex all data) by adding `db.janusgraph.forceDropAndRebuildIndex = true` in application.conf.
+
+!!! Warning "Remove this line once TheHive is started"
+
 #### Pre requisites
 
 Following data is required to restore TheHive database successfully: 
 
-- The database schema (example: `schema.cql`)
-- A backup of the database (example: `backup.tbz`)
+- A backup of the database (`${SNAPSHOT}_${SNAPSHOT_DATE}.tbz`)
 - Keyspace to restore does not exist in the database (or it will be overwritten)
 
 #### Restore 
 
-1. Restore keyspace
+1. Create the keyspace
 
     ```bash
-    cqlsh -u cassandra -p <CASSANDRA_PASSWORD> <IP> -e "source 'schema.cql';"
+    cqlsh -u cassandra -p ${CASSANDRA_PASSWORD} ${CASSANDRA_ADDRESS} -e "
+        CREATE KEYSPACE ${TARGET_KEYSPACE}
+        WITH replication = {'class': 'SimpleStrategy', 'replication_factor': '1'}
+        AND durable_writes = true;"
     ```
 
+    Here it is possible to use a different keyspace name.
+
 2. Unarchive backup files: 
 
     ```bash
-    tar jxf /PATH/TO/backup.tbz -C /tmp/cassandra_backup
+    RESTORE_PATH=$(mktemp -d)
+    tar jxf ${SNAPSHOT}_${SNAPSHOT_DATE}.tbz -C ${RESTORE_PATH}
     ```
 
-3. And restore snapshots files:
+3. Create tables from archives
+
+    The archive contains the table schema. They must be executed in the new keyspace. The schema file are in `${RESTORE_PATH}/var/lib/cassandra/data/${SOURCE_KEYSPACE}/${TABLE}/snapshots/${SNAPSHOT}_${SNAPSHOT_DATE}/schema.cql`
```

Comment:
Not sure about what should be plural here. Does this suggestion make sense :

`The archives contain the table schema. They must be executed in the new keyspace. The schema files are in`

Reply:
one archives, several files and table schemas